frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/components/Tooltip/index.tsx

```diff
@@ -6,44 +6,84 @@ import {
 } from "@blueprintjs/core";
 import { IconNames } from "@blueprintjs/icons";
 import React, { FC } from "react";
-import { UPLOAD_STATUS, VALIDATION_STATUS } from "src/common/entities";
-import { RED } from "src/components/common/theme";
+import {
+  CONVERSION_STATUS,
+  UPLOAD_STATUS,
+  VALIDATION_STATUS,
+} from "src/common/entities";
+import { ORANGE, RED } from "src/components/common/theme";
+import { FAILED_RETURN_TYPE, FailReturn } from "../../utils";
 import { StyledAnchor } from "./style";
 
-const tooltipContent = (error: VALIDATION_STATUS | UPLOAD_STATUS) => {
-  return error === VALIDATION_STATUS.INVALID ? (
-    <span>
-      You must validate your dataset locally before uploading. We provide a
-      local CLI script to do this.{" "}
-      <b>
-        <StyledAnchor
-          href="https://github.com/chanzuckerberg/cellxgene/blob/main/dev_docs/schema_guide.md"
-          target="_blank"
-        >
-          Learn More
-        </StyledAnchor>
-      </b>
-    </span>
-  ) : (
-    <span>There was a problem uploading your file. Please try again.</span>
-  );
+interface Content {
+  color: string;
+  intent: Intent;
+  content: JSX.Element;
+}
+
+const ERROR_TO_CONTENT: { [key: string]: Content } = {
```

Comment:
Create this `ERROR_TO_CONTENT` key value map to keep track of what `FAILED_RETURN_TYPE` and `error` type should render what content